### PR TITLE
Refactoring of the configuration framework to allow autogeneration of config docs 

### LINKED ISSUE #156: refactor config file parsing to allow automatic generation of documentation

### PR DESCRIPTION
**Pull request by wouterdb**
Config is now expected to be defined prior to use. 

To define a config option

`cfg_env = Option("config", "environment", None, "The environment this model is associated with", is_uuid_opt)`

to get the option

`cfg_env.get()`

Major changes:
 - config options are returned as the type given in the definition, not longer as strings
 - '-' and '_' are equivalent in config options
 - server is moved from a file to a directory

fixes #156

### CODE REVIEW

File: src/inmanta/ast/statements/call.py [hunk 1]
```diff
@@ -104,13 +104,13 @@ def execute(self):
         requires = {k: v.get_value() for (k, v) in self.requires.items()}
         try:
             self.function.resume(requires, self.resolver, self.queue_scheduler, self.result)
+            self.done = True
```
Comment (bartv) [inline src/inmanta/ast/statements/call.py]: Is this config related?

Comment (wouterdb) [inline src/inmanta/ast/statements/call.py]: no, seems like a generic fix, my bad.
